.github/workflows/win_build_portable.yml

```diff
@@ -0,0 +1,143 @@
+            libs/ssl.lib
+            libs/openssl_VERSION
+          key: win-openssl-libs-3.1.0-${{ hashFiles('etc/win-ci/build-openssl.ps1') }}-msvc
+          fail-on-cache-miss: true
+      - name: Restore DLLs
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            libs/pcre-dynamic.lib
+            libs/pcre2-8-dynamic.lib
+            libs/iconv-dynamic.lib
+            libs/gc-dynamic.lib
+            libs/ffi-dynamic.lib
+            libs/z-dynamic.lib
+            libs/mpir-dynamic.lib
+            libs/yaml-dynamic.lib
+            libs/xml2-dynamic.lib
+            dlls/pcre.dll
+            dlls/pcre2-8.dll
+            dlls/libiconv.dll
+            dlls/gc.dll
+            dlls/libffi.dll
+            dlls/zlib1.dll
+            dlls/mpir.dll
+            dlls/yaml.dll
+            dlls/libxml2.dll
+          key: win-dlls-${{ hashFiles('.github/workflows/win.yml', 'etc/win-ci/*.ps1') }}-msvc
+          fail-on-cache-miss: true
+      - name: Restore OpenSSL DLLs
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            libs/crypto-dynamic.lib
+            libs/ssl-dynamic.lib
+            dlls/libcrypto-3-x64.dll
+            dlls/libssl-3-x64.dll
+          key: win-openssl-dlls-3.1.0-${{ hashFiles('etc/win-ci/build-openssl.ps1') }}-msvc
+          fail-on-cache-miss: true
+      - name: Restore LLVM
+        uses: actions/cache/restore@v3
+        with:
+          path: llvm
+          key: llvm-libs-16.0.3-msvc
+          fail-on-cache-miss: true
+
+      - name: Set up environment
+        run: |
+          echo "CRYSTAL_LIBRARY_PATH=$(pwd)\libs" >> ${env:GITHUB_ENV}
+          echo "LLVM_CONFIG=$(pwd)\llvm\bin\llvm-config.exe" >> ${env:GITHUB_ENV}
+
+      - name: Build LLVM extensions
+        run: make -f Makefile.win deps
+
+      - name: Build Crystal
+        run: |
+          bin/crystal.bat env
+          make -f Makefile.win -B ${{ inputs.release && "release=1" || "" }}
+
+      - name: Download shards release
+        uses: actions/checkout@v3
+        with:
+          repository: crystal-lang/shards
+          ref: v0.17.3
+          path: shards
+
+      - name: Download molinillo release
+        uses: actions/checkout@v3
+        with:
+          repository: crystal-lang/crystal-molinillo
+          ref: v0.2.0
+          path: shards/lib/molinillo
+
+      - name: Build shards release
+        working-directory: ./shards
+        run: ../bin/crystal.bat build ${{ inputs.release && "--release" || "" }} src/shards.cr
+
+      - name: Gather Crystal binaries
+        run: |
+          mkdir crystal/src
+          mkdir crystal/lib
+          cp .build/crystal.exe crystal/
+          cp shards/shards.exe crystal/
+          cp libs/* crystal/lib/
+          cp dlls/* crystal/
+          cp src/* crystal/src -Recurse
+          cp README.md crystal/
+          cp LICENSE crystal/LICENSE.txt
+          rm crystal/src/llvm/ext/llvm_ext.obj

```

Comment:
Could this make use of `make -f Makefile.win install prefix=crystal` for moving parts of the crystal artifacts? Libs and readme would need to be handled explicitly. But the rest should be identical to the makefile.
There's no Windows makefile for shards, so `shards.exe` needs to be copied manually.